sregistry/main/registry/auth.py

```diff
@@ -16,25 +16,42 @@
     generate_timestamp
 )
 
+from sregistry.logger import bot
+
+def _check_secret(self, secret):
+
```

Comment:
Please add a docstring for what this function does!

Reply:
Hmm, I'm not sure I like how this is implemented - you are passing the client (self) as a parameter which isn't good practice I don't think. In that our functions are to write this to not include self (and then we don't have the helper functions) or to add another function to the client base, I think it might be best to restore to what you originally had.

Reply:
I can also make this a method and add it to the `registry.Client` class implementation

Reply:
Yeah I thought about that, but I think it's too niche a function to provide to all clients.